cartofacade/capi.go

```diff
@@ -57,8 +55,8 @@ type CartoInterface interface {
 	start() error
 	stop() error
 	terminate() error
-	addLidarReading(string, []byte, time.Time) error
-	addIMUReading(string, IMUReading, time.Time) error
+	addLidarReading(string, s.TimedLidarReadingResponse) error
+	addIMUReading(string, s.TimedIMUReadingResponse) error
```

Comment:
Cartographer ingests lidar, imu, and odometer data separately. This means we have to split the movement sensor reading into imu and odometer at some point. This will happen in one of the next PRs in `movementsensorprocess.go`, where we'll rename `addIMUReadingInOnline`/`addIMUReadingInOffline` to `addMovementSensorReadingInOnline`/`addMovementSensorReadingInOffline`, and within those functions we'll split the data into IMU and odometer, which is then fed through the functions above into cartographer.

(Note: `addIMUReadingInOffline` is actually going to be deleted and replaced with a sequential data process)

Reply:
Starting from cartographer, these are the functions that we call to add lidar and imu data: https://github.com/viamrobotics/viam-cartographer/blob/9ad4257d22c99fd02dfc4cfef6bcbe4122e6ffb8/viam-cartographer/src/carto_facade/map_builder.cc#L98-L107

Reply:
One level up is carto facade: https://github.com/viamrobotics/viam-cartographer/blob/9ad4257d22c99fd02dfc4cfef6bcbe4122e6ffb8/viam-cartographer/src/carto_facade/carto_facade.cc#L796-L891

Which on the go side relates to this: https://github.com/viamrobotics/viam-cartographer/blob/9ad4257d22c99fd02dfc4cfef6bcbe4122e6ffb8/cartofacade/carto_facade.go#L68-L109

Those functions are called in `lidarsensorprocess.go` and `movementsensorprocess.go`: 
* https://github.com/viamrobotics/viam-cartographer/blob/9ad4257d22c99fd02dfc4cfef6bcbe4122e6ffb8/sensorprocess/lidarsensorprocess.go#L106-L123
* https://github.com/viamrobotics/viam-cartographer/blob/9ad4257d22c99fd02dfc4cfef6bcbe4122e6ffb8/sensorprocess/movementsensorprocess.go#L133-L147

Those in turn are called in `addIMUReadingInOnline`/...etc., which is where we're going to split the movement sensor data into IMU and odometer data.

Reply:
I think these are all choices we've made though, you can go ahead and merge, not blocking at all, but I would expect a Pose and Twist-like message to be consumed by cartographer itself, since the ros topic that ros wraps cartographer with is odometry http://docs.ros.org/en/noetic/api/nav_msgs/html/msg/Odometry.html mapping to our Position, Orientation, LinearVelocity and AngularVelocity APIs. The answer may be deeper than our facade. I'm not sure.